### PR TITLE
MoveIt + README + Fix Gazebo 

### PR DESCRIPTION
**Pull request by ymollard**
* MoveIt package
* Updated README a bit
* Some Gazebo fixes for melodic
* ROS disconnection in Java file
* Default to iiwa7 in position control (do you want to revert this?)

### CODE REVIEW

**Comment by costashatz** [comment]
Hey! Thanks a lot for this!

I will have a detailed look at it as soon as possible! Some quick comments:

- I really like the MoveIt integration! I had in mind but never found time to do it! Thanks!
- Yes please do not default to IIwa7 and position control. The default should be what it was. Make your own launch files for your specific setup. I am being a bit picky on this mainly because I am in the process of changing these files in order to be able to launch multiple robots at the same time.. See the [namespaces](https://github.com/epfl-lasa/iiwa_ros/tree/namespaces) branch..
- Why do we need the 'commanding' publisher?

Thanks again!

**Comment by ymollard** [comment]
OK for defaulting back to iiwa14 (also in MoveIt), I no longer have the possibility to test on a real iiwa though,  I don't know if everything will be OK especially with MoveIt.

> Why do we need the 'commanding' publisher?
Do you know another way to detect STOP1 (robot stopped by the safety controller) or a disconnected robot?

There is also a dangerous behavior: if the robot leaves the planned path at some point and goes into STOP1, at resuming the ROS controller will resume it according to its new position, that triggers a high velocity motion to correct the position (and causes another STOP1 depending the safety conf). My workaround is to switch off the ROS stack if commanding become False. There are probably some safety checks to be done in the controller to prevent this.

**Comment by costashatz** [comment]
> There is also a dangerous behavior: if the robot leaves the planned path at some point and goes into STOP1, at resuming the ROS controller will resume it according to its new position, that triggers a high velocity motion to correct the position (and causes another STOP1 depending the safety conf). My workaround is to switch off the ROS stack if commanding become False. There are probably some safety checks to be done in the controller to prevent this.

OK. Now I get it.. This is related to MoveIt.. I never used the code with MoveIt!

> OK for defaulting back to iiwa14 (also in MoveIt), I no longer have the possibility to test on a real iiwa though, I don't know if everything will be OK especially with MoveIt.

I will test both on Iiwa7 and Iiwa14 and make sure that everything works fine.. Please revert to the previous launch files and I will start the testing and reviewing of your code as soon as possible..

Thanks again!

**Comment by ymollard** [comment]
> OK. Now I get it.. This is related to MoveIt.. I never used the code with MoveIt!

It might be yes. The way MoveIt resumes after emergency stop should be a TODO.
But also maybe the controller should perform some sanity checks according to maximum velocities and accelerations in iiwa specs.

Have you ever executed `JointTrajectories` trajectories with this package or do you always run real-time control?  

> Please revert to the previous launch files and I will start the testing and reviewing of your code as soon as possible..

This is reverted. iiwa7 and 14 both work at least in with MoveIt + Gazebo.

**Comment by costashatz** [comment]
> But also maybe the controller should perform some sanity checks according to maximum velocities and accelerations in iiwa specs.

This is already being done..

> Have you ever executed `JointTrajectories` trajectories with this package or do you always run real-time control?

I never used joint trajectories.. Always directly controlled the robots..

> This is reverted. iiwa7 and 14 both work at least in with MoveIt + Gazebo.

Thanks! I will try to have a look as soon as possible..

File: README.md [hunk 1]
```diff
@@ -106,18 +106,37 @@ cd /path/to/ros_workspace
...
 --------------
 
 ### Bringup iiwa_driver
 
 **Control IIWA with FRI**
 
-```sh
-roslaunch iiwa_driver iiwa_bringup.launch
-```
+1. Make sure your Linux/ROS laptop is connected on the KONI Ethernet port and has IP `192.170.10.100` mask `255.255.255.0`.
```
Comment (prachandabhanu) [inline README.md]: I have the following questions regarding the IP config:
1. I am not able to ping the robot with [IPv4](https://github.com/epfl-lasa/iiwa_ros/blob/c75f970cae69eb89a4c1e69f63c83824c549dc3d/iiwa_driver/config/iiwa.yaml#L3) and Robot IP as mentioned here.
2. Do we need to change the IP [here](https://github.com/epfl-lasa/iiwa_ros/blob/c75f970cae69eb89a4c1e69f63c83824c549dc3d/iiwa_driver/java/FRIOverlay.java#L36) to Linux /ROS Master
3. I connected a monitor to KONI and checked the IPv4 configuration of two adopters first one 172.31.1.147 (Here I connect my Sunrise to sync the and load the app, etc) and second IPv4 config 192.168.0.2. In my guess, the second adopter is KONI IPv4 config.
4. Also, I am confused about whether to change the KONI IP in setup.cat. Currently, it is set to 192.168.0.7 

Thank you

**Comment by ymollard** [comment]
> This is already being done..
I'm not exactly sure, but these should work as reproducible steps:
1. From the MoveIt GUI, plan some motion
2. Press STOP on the Smartpad while the motion is performing
3. Switch to T1 and manually move your manipulator in a different arm configuration with single axis buttoins
4. (MoveIt and the driver should be still running...)
5. Switch to AUT again and press PLAY
6. The controller will quickly send the manipulator back to where MoveIt stopped

**Comment by costashatz** [comment]
@ymollard I finally found some time to look into this.

I am running MoveIt! on the real robot and I am getting the following: `Controller is taking too long to execute trajectory (the expected upper bound for the trajectory execution was 3.688350 seconds). Stopping trajectory.`

This is in position control. Any ideas? Thanks!

**Comment by costashatz** [comment]
@ymollard No worries I found how to by-pass this issue. I am doing some last checks and then I will merge! Thanks a lot!

File: iiwa_moveit/launch/trajectory_execution.launch.xml [hunk 2]
```diff
@@ -8,7 +8,7 @@
     <param name="moveit_manage_controllers" value="$(arg moveit_manage_controllers)"/>
     
     <!-- When determining the expected duration of a trajectory, this multiplicative factor is applied to get the allowed duration of execution -->
-    <param name="trajectory_execution/allowed_execution_duration_scaling" value="1.2"/> <!-- default 1.2 -->
+    <param name="trajectory_execution/allowed_execution_duration_scaling" value="3.0"/> <!-- default 1.2 -->
```
Comment (ymollard) [inline iiwa_moveit/launch/trajectory_execution.launch.xml]: Re: https://github.com/epfl-lasa/iiwa_ros/pull/10#issuecomment-589620486
I used the same workaround at some point, however that should not be the actual fix. Trajectories are slowed down by something to be found.

**Comment by ymollard** [comment]
> > But also maybe the controller should perform some sanity checks according to maximum velocities and accelerations in iiwa specs.
> This is already being done..

Did you manage to reproduce this? I don't remember how I ended up with these issues, maybe by keeping the smartpad's play button pressed after resuming from emergency stop. It looks like keeping PLAY pressed has a peculiar meaning in the Java app.

File: iiwa_moveit/launch/trajectory_execution.launch.xml (hunk 2 above)
Comment (costashatz) [reply iiwa_moveit/launch/trajectory_execution.launch.xml]: You are right about this, but MoveIt! is not really something I will actively be using. The community is free to improve/fix this..

**Comment by costashatz** [comment]
> Did you manage to reproduce this?

No I couldn't. But having one topic more does not really hurt the performance, so I left the `commanding` topic there..

> Does "long press play" mean anything peculiar in Kuka Java language?

No idea to be honest. I am not really an expert of KUKA equipement.

Thanks a lot again!

**Pull request merged**